### DISCUSSION TITLE
Android Automotive OS

### DISCUSSION THREAD

**Discussion by dpetrecki** [organicmaps/organicmaps#4252] [Q&A] [answered]
Hello, can you please share a manual, how to build Organic Maps for AAOS? I've tried the official web APK but it crashes after few seconds. I've tried to build fdroid release and sign it with the platform key of my Android build but it crashes in the same way. I've seen it's possible (https://github.com/organicmaps/organicmaps/pull/3806#issuecomment-1301933222) but I have no idea, how to make it done.

**Comment by biodranik** [comment, marked as answer]
AA/AAOS are still in progress. Do you want to help us with the development? Check `aa` branch.

**Reply by dpetrecki** [reply]
Hello and thank you for your answer. I've seen in the linked PR that the `aa` branch is focused on AA and it's difficult to have both AA and AAOS supported in the single application. There is also an information "On master branch it works perfectly" related to AAOS.
Is there a way to start it on AAOS (without crashing few seconds later) to even start working on it? Maybe @AndrewShkrob can help?

**Reply by AndrewShkrob** [reply]
@dpetrecki It's not possible to work on AAOS before we finish AA. AAOS and AA use the same library and we are focused on AA now because AAOS is used only in ~30 car models (according to [wiki](https://en.wikipedia.org/wiki/Android_Automotive)). It shouldn't be a problem to add AAOS support later when we finish with AA. 
> There is also an information "On master branch it works perfectly" related to AAOS.

I meant that it at least builds and runs 😄
I didn't check how it works. AND it's not a true AAOS app. It's just our regular application. It may not be possible to run it on a real hardware, only on emulators. If you have a car with AAOS you can try to check its settings whether there is a checkbox that allows running android applications.

If you want to help us with AA development, you can check this issue https://github.com/organicmaps/organicmaps/issues/4096.
There is a list of features needed to be implemented.

**Reply by dpetrecki** [reply]
Thank you. TBH, ~30 car models is a very optimistic thinking. Only some of them are really buyable today. I don't have a car equipped with AAOS. I have an easy-to-buy ARM board and I'm trying to create a car-agnostic, open-source AAOS without GAS. I'm mostly focused on CAN-sourced VHAL but I don't like a big green map placeholder on the main screen. When I found your comment in the AA topic, I thought Orgamic Maps may be my navigation provider but I'm not an Android developer so I don't think I can help you with the development. I'm sorry about that...
